### ISSUE TITLE
Lofreq MAFs lack allele counts

### ISSUE THREAD

**Issue by rdmorin** [LCR-BCCRC/lcr-modules#138]
I noticed that the values that we can use to recalculate VAF (t_depth, t_ref_count, t_alt_count) are present in Strelka MAFs but missing from the MAF files from Lofreq VCFs. I am not sure (yet) if Mutect will also be affected. We should ensure this information is consistently present in all MAFs, if possible.

**Comment by Kdreval** [comment]
Hm, I checked the Lofreq VCFs and the counts are in the vcf outputs, but not the tumour depth. Also, doesn't seem like these counts travel to the mafs in general (not only missing from the columns you specified, but also from other columns, too). I think what I can do is to retrieve this information from the vcf file, and then populate columns in the maf file. I will think about this, and also see how to get the tumour depth from the lofreq outputs.

**Comment by Kdreval** [comment]
So the `t_depth` should be coming from the `DP` column of vcf, which is there for the Lofreq outputs. I think the reason `t_ref_count `and `t_alt_count` are not populated is because Lofreq vcfs are missing the `AD` (or `RD`) column information. Similar to AF->VAF, I will need to work on the format of Lofreq outputs. I think once that is done, the vcf2maf conversion should go properly.

**Comment by rdmorin** [comment]
I suggest you start by figuring out how/why this works out of the box for Strelka VCFs. It may be easier to modify the Lofreq VCFs to just "work" with vcf2maf.

**Comment by rdmorin** [comment]
I posted this before seeing your second comment. Yes, you're on the right track.

**Comment by Kdreval** [comment]
I looked further and found that vcf2maf does not support the `DP4` column, where Lofreq outputs counts. To make it working with vcf2maf, I think I will add together the counts for ref-forward bases and ref-reverse from lofreq output to get the value of total ref counts, and will create a new column so vcf2maf can recognize and use it for `t_ref_count`.

**Comment by Kdreval** [comment]
I did more investigation and realized it is a common problem. After inspecting the vcf2maf code, I found that the `DP4` column _is_ actually supported in the latest version of `vcf2maf`, but not in the way it is output by lofreq. The difference is that it needs to be in the FORMAT, not INFO column of vcf. I followed issues # 57 from lofreq git, and # 167 from vcf2maf git. I came up with the solution of adding the `FORMAT` column to the lofreq vcfs, and `TUMOR` column with values for the `FORMAT` column. Once modified in this way, vcf2maf populates the missing columns.
